lib/msal-browser/src/config/Configuration.ts

```diff
@@ -156,6 +158,11 @@ export function buildConfiguration({ auth: userInputAuth, cache: userInputCache,
         allowRedirectInIframe: false
     };
 
+    // build authority string based on auth params - azureCloudInstance is prioritized if provided
+    if (userInputAuth && userInputAuth.azureCloudInstance) {
+        userInputAuth.authority = `${Authority.getAzureCloudInstanceUrl(userInputAuth.azureCloudInstance)}/${Constants.DEFAULT_AUTHORITY_TENANT}/`;
```

Comment:
Can we do this transformation in the request generation instead?

Reply:
Also is this only supported for `/common`? If so we should validate that the user provided `authority` is also `/common`

Reply:
This is mostly because we may not have tenant info if the user provides only an `enum`. I can check the request generation part.

Reply:
Dotnet has a config option for `tenantId`, which we don't have, and that can be combined with these enum to build the authority: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/9e1a300564efb5ad2d031d5149012c8e47749e3b/src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs#L526-L601

Reply:
Imo, having `authority` and `azureCloudInstance` (and potentially `tenantId`) is strange, and the library would have to be opinionated about either which option takes precedence (or throwing errors if both are provided).

Reply:
This may change the public interface. I have seen changing the type, even if it supports the old type, will break apps, depending on their import of types.

Reply:
> Also is this only supported for `/common`? If so we should validate that the user provided `authority` is also `/common`

I do not think so. 

* We do not have a tenant configuration separately in our library.I am not in favor of introducing this in v2 as is since it is over populated already w.r.t configuration options for `authority`. Redesigning `authority interface` in `V3` can be a better option to introduce `tenantId` configurability IMO.  
* User provided `authority` should be able to provide any `tenantId`. Do not intend to change that. 
* `AzureCloudInstance` is intended to just provide convenience for initiating a specific cloud for certain users (especially sovereign clouds) since docs on them seem to be sparse and unclear. Hence made a decision to default the `tenantId` to `common`.

One option is to check if the user provided `authority` matches the user choice for `azureCloudInstance` if provided, and append the `tenantId` if it is provided. This does not fix all use cases and complicated the user interface.